### PR TITLE
Flipping ratio correction

### PR DESCRIPTION
**Pull request by AndreiSavici**
**Description of work.**

This is a new algorithm to correct for flipping ratio. It multiplies MDEvents by  F/(F-1) and 1/(F-1). F can be a function of some log values

**To test:**
See documentation example. 

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by aybamidele** [comment]
Could you point me to the documentation for this?

**Comment by AndreiSavici** [comment]
https://inis.iaea.org/collection/NCLCollectionStore/_Public/16/076/16076003.pdf is the Scarpf paper (see page 87). The last formulas were scaled up, so the intensity is conserved. The link for the orther paper is in the help file. Let me know if you have issues retrieving it.

File: Framework/MDAlgorithms/src/FlippingRatioCorrectionMD.cpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  }
+  for (auto &boxe : boxes) {
+    DataObjects::MDBox<MDE, nd> *box =
+        dynamic_cast<DataObjects::MDBox<MDE, nd> *>(boxe);
+    if (box) {
+      typename std::vector<MDE> &events = box->getEvents();
+      size_t ic(events.size());
+      auto it = events.begin();
+      auto it_end = events.end();
+      for (; it != it_end; it++) {
```
Comment (peterfpeterson) [inline Framework/MDAlgorithms/src/FlippingRatioCorrectionMD.cpp]: Couldn't this be a range-based `for` loop?

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Lots of changes in your inner loop.

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**